kmmsdk/src/androidMain/kotlin/com/lokate/kmmsdk/AndroidEstimoteBeaconScanner.kt

```diff
@@ -0,0 +1,142 @@
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.launch
+
+class AndroidEstimoteBeaconScanner(appId: String, appToken: String) : BeaconScanner {
+    private val cloudCredentials = EstimoteCloudCredentials(appId, appToken)
+    private val proximityObserver =
+        ProximityObserverBuilder(applicationContext, cloudCredentials)
+            .withLowLatencyPowerMode()
+            .withTelemetryReportingDisabled()
+            .withEstimoteSecureMonitoringDisabled()
+            .withAnalyticsReportingDisabled() // we will use our own analytics
+            .onError {
+                    error ->
+                Log.e("AndroidEstimoteBeaconScanner", "Error: ${error.message}")
+                stopScanning()
+            }
+            .build()
+    private var observationHandler: ProximityObserver.Handler? = null
+
+    companion object {
+        private val mainJob = SupervisorJob()
+        private val coroutineScope = CoroutineScope(Dispatchers.IO + mainJob)
+    }
+
+    private var running: Boolean = false
+
+    private val regions = mutableListOf<ProximityZone>()
+    private val beaconFlow: MutableSharedFlow<BeaconScanResult> = MutableSharedFlow()
+
+    private val scanResults = mutableSetOf<BeaconScanResult>()
+
+    override fun startScanning() {
+        if (running) {
+            Log.e("AndroidEstimoteBeaconScanner", "Already scanning")
+            return
+        }
+        if (regions.isEmpty()) {
+            setRegions(listOf())
+        }
+
+        Log.d("AndroidEstimoteBeaconScanner", "Starting scanning")
+
+        observationHandler =
+            with(proximityObserver) {
+                running = true
+                flowEmitter()
+                startObserving(regions)
+            }
+    }
+
+    override fun stopScanning() {
+        if (!running) {
+            Log.e("AndroidEstimoteBeaconScanner", "Not scanning anyways")
+            return
+        }
+        observationHandler!!.stop()
+        mainJob.cancel()
+        running = false
+    }
+
+    override fun setRegions(regions: List<LokateBeacon>) {
+        if (running) {
+            Log.d("AndroidEstimoteBeaconScanner", "Already running!")
+            return
+        }
+        if (regions.isEmpty()) {
+            Log.d("AndroidEstimoteBeaconScanner", "No regions to scan")
+            return
+        }
+
+        Log.d("AndroidEstimoteBeaconScanner", "Setting regions")
+
+        this.regions.clear()
+        this.regions.addAll(
+            regions
+                .map { beacon ->
+                    ProximityZoneBuilder()
+                        .forTag(beacon.proximityUUID.uppercase() + ':' + beacon.major + ':' + beacon.minor)
+                        .inCustomRange(Double.MAX_VALUE)
```

Comment:
https://github.com/lokateapp/LokateMobileSDK/pull/7/files#r1538288366

We should pass the distance estimote sdk does not allow us to get the accuracy field

Reply:
yes, I was intentionally passing the distance before but forgot that during rebase.

Reply:
I can update it on the iOS PR if it is okay